plugin/pkg/scheduler/algorithm/priorities/priorities.go

```diff
@@ -273,3 +273,74 @@ func fractionOfCapacity(requested, capacity int64) float64 {
 	}
 	return float64(requested) / float64(capacity)
 }
+
+type NodePreferAvoid struct {
+	controllerLister algorithm.ControllerLister
+}
+
+func NewNodePreferAvoidPriority(controllerLister algorithm.ControllerLister) algorithm.PriorityFunction {
+	nodePreferAvoid := &NodePreferAvoid{
+		controllerLister: controllerLister,
+	}
+	return nodePreferAvoid.CalculateNodePreferAvoidPriority
+}
+
+func (npa *NodePreferAvoid) CalculateNodePreferAvoidPriority(pod *api.Pod, nodeNameToInfo map[string]*schedulercache.NodeInfo, nodeLister algorithm.NodeLister) (schedulerapi.HostPriorityList, error) {
+	var score int
+	nodes, err := nodeLister.List()
+	if err != nil {
+		return nil, err
+	}
+
+	result := []schedulerapi.HostPriority{}
+
+	rcs, err := npa.controllerLister.GetPodControllers(pod)
```

Comment:
GetPodControllers() only gives you ReplicationControllers. You should also do GetPodReplicaSets() (see how this is done in CalculateSpreadPriority() in plugin/pkg/scheduler/algorithm/priorities/selector_spreading.go). Once we have controllerRef we can just use that, to find the pod's controller, so please also add

// TODO: Once we have controllerRef, use it to find controller for the pod.